src/utils.rs

```diff
@@ -496,5 +506,43 @@ where
     let descriptor = wallet_opts.descriptor.as_str();
     let change_descriptor = wallet_opts.change_descriptor.as_deref();
     let wallet = Wallet::new(descriptor, change_descriptor, network, database)?;
+
+    #[cfg(feature = "hardware-signer")]
+    let wallet = add_hardware_signers(wallet, network)?;
+
+    Ok(wallet)
+}
+
+/// Add hardware wallets as signers to the wallet
+#[cfg(feature = "hardware-signer")]
+fn add_hardware_signers<D>(wallet: Wallet<D>, network: Network) -> Result<Wallet<D>, Error>
+where
+    D: BatchDatabase,
+{
+    let mut wallet = wallet;
+    //    let chain = HWIChain::from(network);
```

Comment:
This doesn't work so can be removed..

Reply:
I still think this line should work, and I could remove the match statement instead. But I couldn't figure out how to make it work.

Reply:
Ideally, it should. But `HWIChain` doesn't implement `From<Network>`. Thats why its not working.. Probably worth a PR into HWI lib..